pkg/storage/quota_pool.go

```diff
@@ -0,0 +1,81 @@
+package storage
+
+import (
+	"math"
+	"sync"
+
+	"golang.org/x/net/context"
+)
+
+const (
+	leaderProposalQuota   = 1000
+	followerProposalQuota = math.MaxInt64 / 2
+)
+
+type quotaPool struct {
+	c chan int64
+
+	mu    sync.Mutex
```

Comment:
This code was lifted from `grpc`. I haven't thought through the difficulties with using atomics here.